plugins/repository-s3/build.gradle

```diff
@@ -265,6 +282,46 @@ if (useFixture) {
   }
 }
 
+// STS (Secure Token Service)
+if (useFixture) {
+  testFixtures.useFixture(':test:fixtures:s3-fixture', 's3-fixture-with-sts')
+  tasks.register("yamlRestTestSTS", RestIntegTestTask.class) {
+    description = "Runs tests with the STS (Secure Token Service)"
+    dependsOn('bundlePlugin')
+
+    SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+    SourceSet yamlRestTestSourceSet = sourceSets.getByName(InternalYamlRestTestPlugin.SOURCE_SET_NAME)
+    setTestClassesDirs(yamlRestTestSourceSet.getOutput().getClassesDirs())
+    setClasspath(yamlRestTestSourceSet.getRuntimeClasspath())
+    systemProperty 'tests.rest.blacklist', [
+      'repository_s3/10_basic/*',
+      'repository_s3/20_repository_permanent_credentials/*',
+      'repository_s3/30_repository_temporary_credentials/*',
+      'repository_s3/40_repository_ec2_credentials/*',
+      'repository_s3/50_repository_ecs_credentials/*'
+    ].join(",")
+  }
+  tasks.named("check").configure { dependsOn("yamlRestTestSTS") }
+
+  testClusters.matching { it.name == "yamlRestTestSTS" }.configureEach {
+    plugin tasks.bundlePlugin.archiveFile
+
+    setting 's3.client.integration_test_sts.endpoint', { "${-> fixtureAddress('s3-fixture', 's3-fixture-with-sts', '80')}" }, IGNORE_VALUE
+    systemProperty 'com.amazonaws.sdk.stsMetadataServiceEndpointOverride',
+      { "${-> fixtureAddress('s3-fixture', 's3-fixture-with-sts', '80')}/assume-role-with-web-identity" }, IGNORE_VALUE
+
+    File awsWebIdentityTokenExternalLocation = file('src/test/resources/aws-web-identity-token-file')
+    // The web identity token can be read only from the plugin config directory because of security restrictions
+    // Ideally we would create a symlink, but extraConfigFile doesn't support it
```

Comment:
Maybe the es-delivery team can expose a way to use LinkOption.NOFOLLOW_LINKS with extraConfigFile? Otherwise were are not really testing the documented setup.

Reply:
ie something like :
```java
                if (Files.isSymbolicLink(from.toPath())) {
                    Files.copy(from.toPath(), dst, LinkOption.NOFOLLOW_LINKS);
                } else {
                    Files.copy(from.toPath(), dst, StandardCopyOption.REPLACE_EXISTING);
                }
```java

in `ElasticsearchNode.copyExtraConfigFiles()` should do the trick if the symlink is created before, but I'm not sure what the implications for the build could be.

Reply:
I also was thinking about that option. Let me get in touch with the delivery team and see if we can get it done.

Reply:
Can we follow up on this in a separate PR? I need to work with the delivery team to implement the support for symlinks which will take some time. Meanwhile, copying the file shouldn't be a blocker.

Reply:
> Can we follow up on this in a separate PR

Sure, let's not block this change for this.